api/src/services/fields.ts

```diff
@@ -928,13 +943,19 @@ export class FieldsService {
 			}
 		}
 
+		const createIndexes: { unique: boolean }[] = [];
+
 		if (field.schema?.is_primary_key) {
 			column.primary().notNullable();
 		} else if (!existing?.is_primary_key) {
 			// primary key will already have unique/index constraints
 			if (field.schema?.is_unique === true) {
 				if (!existing || existing.is_unique === false) {
-					column.unique({ indexName: this.helpers.schema.generateIndexName('unique', collection, field.field) });
+					if (tryNonBlocking) {
+						createIndexes.push({ unique: true });
+					} else {
+						column.unique({ indexName: this.helpers.schema.generateIndexName('unique', collection, field.field) });
+					}
```

Comment:
Should we simplify this by always building an array of indexes to add? The logic in `createIndex` schema helper seems equivalent with `tryNonBlocking` as false. Same goes for regular index creation below.

Reply:
The reason why I havent done so was because of how knex column updates work here. If I always fill the index array then I also need to figure out IF anything else has changed on the field in this function and prevent `column.alter();` because that errors otherwise. So mainly because I wasnt sure how to nicely check for changes besides an index creation 🤔